src/migration.ts

```diff
@@ -0,0 +1,269 @@
+import * as fs from "fs";
+import {Response} from "express";
+
+var myLogFileStream = fs.createWriteStream("./console.log");
+var myConsole = new console.Console(myLogFileStream, myLogFileStream);
+
+let checkpoint = '';
+export async function beginMigration(response: Response) {
+
+    console.log("");
+    console.log("");
+    console.log("The Great Wildebeest Migration begins....");
+    console.log("");
+    console.log("...3");
+    console.log("......2");
+    console.log(".........1");
+    console.log("............ GO!");
+    console.log("");
+
+    await MongoDataSource.initialize();
+    await  PostgresDataSource.initialize();
+
+    const cdRepo = MongoDataSource.getRepository(ClinicalDonor);
+    const donors = await cdRepo.find();
+
+    for (const donor of donors) {
+        console.log("");
+        console.log(" ------------ Migrating ids for  donor: " + donor.id+" Program Id: "+donor.programId+" -Submitter Id: "+donor.submitterId+" -------------");
+        console.log("");
+        try {
+            await migrateDonors(donor);
+            await migrateBiomarkers(donor);
+            await migrateComorbidity(donor);
+            await migrateExposure(donor);
+            await migrateFamilyHistory(donor);
+            await migrateFollowUps(donor);
+            await migratePrimaryDiagnosis(donor);
+            await migrateTreatments(donor);
+            await migrateSpecimens(donor);
+            await migrateSamples(donor);
+        } catch (e) {
+            console.log("Oops, Croc attack! - migration for donor failed."+" -Checkpoint: "+checkpoint+" -Program Id: "+donor.programId+" -Submitter Id: "+donor.submitterId);
+            console.log("Problem: "+e.message);
+            await MongoDataSource.getRepository(FailedMigrations).save(donor);
+            continue;
+        }
+    }
+
+    console.log("");
+    console.log('Wildebeest migrated!');
+    response.send;
+}
+
+
+async function migrateDonors(donor: ClinicalDonor){
+    checkpoint = 'Donor: '+donor.donorId;
+
+    const dnRepo = PostgresDataSource.getRepository(Donor);
+    const submitterDonorId = donor.submitterId?.toString();
+    const programId = donor.programId;
+    const idDonor = new Donor();
+    idDonor.submitterDonorId = submitterDonorId || '-';
+    idDonor.programId = programId || '-';
+    idDonor.entityId = donor.donorId?.toString() || '-';
+    idDonor.entityType = "donor";
+    await dnRepo.save(idDonor);
+
+    console.log("donor id "+ idDonor.entityId + " of clinical donor "+donor.id + " migrated.");
+
+}
+
+async function migrateBiomarkers(donor: ClinicalDonor){
+    const bmRepo = PostgresDataSource.getRepository(Biomarker);
+        const submitterDonorId = donor.submitterId?.toString();
+        const programId = donor.programId;
+    for (const bm of donor.biomarker) {
+        checkpoint = 'Biomarker: '+bm.biomarkerId;
+
+        const biomarker = new Biomarker();
+        biomarker.submitterDonorId = submitterDonorId || '-';
```

Comment:
May want to define `'-'` once as a constant since it's reused everywhere
Then if the placeholder ever changes you can just update the variable